src/tests/class_tests/openms/source/AASequence_test.cpp

```diff
@@ -1278,6 +1337,15 @@ START_SECTION([EXTRA] testing terminal modifications)
   TEST_EXCEPTION(Exception::InvalidValue, AASequence::fromString("DFPIANGEM(UniMod:10)R.")) // not just any M is allowed
   TEST_EQUAL( AASequence::fromString(".DFPIANGEM.(UniMod:10)"), AASequence::fromString(".DFPIANGEM.(UniMod:10)") )
   TEST_EQUAL( AASequence::fromString(".DFPIANGEM(UniMod:10)"), AASequence::fromString(".DFPIANGEM.(UniMod:10)") )
+  TEST_EQUAL( AASequence::fromString(".DFPIANGEM.c[-30]"), AASequence::fromString(".DFPIANGEM.(UniMod:10)") )
+  TEST_EQUAL( AASequence::fromString(".DFPIANGEM.c[-29.99]"), AASequence::fromString(".DFPIANGEM.(UniMod:10)") )
+
+  // TO FIX: this is an ambiguous case, toBracketString produces the "absolute"
+  // mass of the n terminal "AA" but its mass is negative, therefore when
+  // parsing it would be interpreted as a delta mass and its off by one!
+  TEST_EQUAL( AASequence::fromString(".DFPIANGEM(UniMod:10)").toBracketString(true), "DFPIANGEMc[-13]") // absolute
+  TEST_STRING_SIMILAR( AASequence::fromString(".DFPIANGEM(UniMod:10)").toBracketString(false), "DFPIANGEMc[-12.9900659681]") // absolute
+  TEST_EQUAL( AASequence::fromString(".DFPIANGEM(UniMod:10)").toBracketString(true, true), "DFPIANGEMc[-30]") // relative
```

Comment:
Why does this problem occur?

UniMod 10 is a "Cyanogen bromide (CNBr) cleavage converts the C-term Met to either homoserine or homoserine lactone, depending on pH." so it does not actually modify the C-terminus, it modifies the amino acid, but only if located at the C-terminus. So the modification in that sense is *not* a c-terminal modification, but a modification of the C-terminal amino acid. Otherwise, of course, we would never have a negative absolute mass of anything, that is simply not possible. However in this case its the mass of the C-terminus + the mass of the final M = positive whereas the mass of the C-terminus alone is negative.

http://www.unimod.org/modifications_view.php?editid1=10